src/template.ts

```diff
@@ -1,97 +1,112 @@
+import type { InsightsClient, InsightsEvent, Init } from 'search-insights';
+import { devDependencies, version } from '../package.json';
+
 const log = require('logToConsole');
 const createArgumentsQueue = require('createArgumentsQueue');
 const injectScript = require('injectScript');
 const queryPermission = require('queryPermission');
 const setInWindow = require('setInWindow');
 const copyFromWindow = require('copyFromWindow');
-const makeInteger = require('makeInteger');
-const getType = require('getType');
-const Math = require('Math');
-const Object = require('Object');
+const makeInteger: GtmMakeInteger = require('makeInteger');
+const getType: GtmGetType = require('getType');
+const Math: GtmMath = require('Math');
+const Object: GtmObject = require('Object');
 
-const TEMPLATE_VERSION = '1.6.6';
 const INSIGHTS_OBJECT_NAME = 'AlgoliaAnalyticsObject';
-const INSIGHTS_LIBRARY_URL =
-  'https://cdn.jsdelivr.net/npm/search-insights@2.14.0';
+const INSIGHTS_LIBRARY_URL = `https://cdn.jsdelivr.net/npm/search-insights@${devDependencies['search-insights']}`;
 
 const MAX_OBJECT_IDS = 20;
 const MAX_FILTERS = 10;
 const COMMA_REPLACEMENT = 'PRESERVED_COMMA_HERE';
 
-const aa = createArgumentsQueue('aa', 'aa.queue');
+const aa: InsightsClient = createArgumentsQueue('aa', 'aa.queue');
 
 function isInitialized() {
   return !!copyFromWindow(INSIGHTS_OBJECT_NAME);
 }
 
-function replaceEscapedCommas(str) {
+function replaceEscapedCommas(str: string) {
   while (str.indexOf('\\,') !== -1) {
     str = str.replace('\\,', COMMA_REPLACEMENT);
   }
   return str;
 }
 
-function replacePlaceholderText(str) {
+function replacePlaceholderText(str: string) {
   while (str.indexOf(COMMA_REPLACEMENT) !== -1) {
     str = str.replace(COMMA_REPLACEMENT, ',');
   }
   return str;
 }
 
-function splitPreservingEscapedCommas(str) {
+function splitPreservingEscapedCommas(str: string) {
   // GTM does not support regexes in custom templates, so we need to use straight text replacements instead
   return replaceEscapedCommas(str).split(',').map(replacePlaceholderText);
 }
 
-function formatValueToList(value) {
-  const type = getType(value);
-  if (type === 'array') {
+function formatValueToList<T>(value: T | T[]): T[] {
+  if (isArray(value)) {
     return value;
-  } else if (type === 'number') {
+  } else if (isNumber(value)) {
     return [value];
-  } else if (type === 'string') {
-    return splitPreservingEscapedCommas(value);
+  } else if (isString(value)) {
+    return splitPreservingEscapedCommas(value) as T[];
   } else {
-    return null;
+    return [];
```

Comment:
This is one of two actual changes. The type checking caught a subtle error. Already a win.